build/index.js

```diff
@@ -320,6 +317,10 @@ async function main() {
         logMain(`Command '${command}' failed: ${error}\n`);
         return 1;
     }
+
+    if (command === "clean") {
```

Comment:
I had to move it down - otherwise we would _first_ install the dependencies needed to clean some sub packages (i.e. `shx`), _then_ we would clean the root package (removing them again!), _then_ we would try to use them.

Found this issue while testing everything a bit, it basically causes all the _custom_ clean scripts some (very few, really) packages have to fail.

Reply:
Right. Deps are installed when build... And there's a implicit dependency for you. I think we should *really* refactor, revamp and even spin off this build script, making it an all-powerful task runner for monorepos called *baut*. It's becoming increasingly clear that we will need to modularize and refactor at some point, and that point is becoming increasingly close as we find this kind of hacks needed to make it just work.

Reply:
Or _baumeister_? Anyway, we can modularise this even if we just stick to _build_ for now - there's already _gyp_ for the hyper generic monorepo tool, I think we want to be specific rather than generic.

As for this hack, there's two problems: The script actually somewhat elegantly _only_ deals with packages, which includes _build_ itself. But we do _not_ treat the root package like a package, and the root dependencies aren't something the dependency tree covers either (cause it introduces dependencies even for the clean scripts).

The other problem, that @alexandrosFarhat actually found: It doesn't do anything here! This code would need to be moved _before_ the `return 0` in the try block above. I wonder why the linter didn't complain...